docs/pages/bdd-references/when.md

```diff
@@ -56,117 +56,117 @@ not met with the infrastructure plan.
 >
 > ▪
-[When](#){: .p-1 .text-red-200} 
-its 
-[property](#){: .p-1 .text-green-200 .fw-700} 
+[When](#){: .p-1 .text-red-200}
+its
+[property](#){: .p-1 .text-green-200 .fw-700}
 has
-[something](#){: .p-1 .text-blue-100 .fw-700} 
+[something](#){: .p-1 .text-blue-100 .fw-700}
 >
 > ▪
-[When](#){: .p-1 .text-red-200} 
-its 
-[property](#){: .p-1 .text-green-200 .fw-700} 
+[When](#){: .p-1 .text-red-200}
+its
+[property](#){: .p-1 .text-green-200 .fw-700}
 includes
-[something](#){: .p-1 .text-blue-100 .fw-700} 
+[something](#){: .p-1 .text-blue-100 .fw-700}
 >
 > ▪
-[When](#){: .p-1 .text-red-200} 
-its 
-[property](#){: .p-1 .text-green-200 .fw-700} 
+[When](#){: .p-1 .text-red-200}
+its
+[property](#){: .p-1 .text-green-200 .fw-700}
 consists
-[something](#){: .p-1 .text-blue-100 .fw-700} 
+[something](#){: .p-1 .text-blue-100 .fw-700}
 >
 > ▪
-[When](#){: .p-1 .text-red-200} 
-its 
-[property](#){: .p-1 .text-green-200 .fw-700} 
+[When](#){: .p-1 .text-red-200}
+its
+[property](#){: .p-1 .text-green-200 .fw-700}
 is
 "
-[some string with spaces](#){: .p-1 .text-blue-100 .fw-700} 
+[some string with spaces](#){: .p-1 .text-blue-100 .fw-700}
 "
 >
 > ▪
-[When](#){: .p-1 .text-red-200} 
-its 
-[property](#){: .p-1 .text-green-200 .fw-700} 
+[When](#){: .p-1 .text-red-200}
+its
+[property](#){: .p-1 .text-green-200 .fw-700}
 has
 "
-[some string with spaces](#){: .p-1 .text-blue-100 .fw-700} 
+[some string with spaces](#){: .p-1 .text-blue-100 .fw-700}
 "
 >
 > ▪
-[When](#){: .p-1 .text-red-200} 
-its 
-[property](#){: .p-1 .text-green-200 .fw-700} 
+[When](#){: .p-1 .text-red-200}
+its
+[property](#){: .p-1 .text-green-200 .fw-700}
 includes
 "
-[some string with spaces](#){: .p-1 .text-blue-100 .fw-700} 
+[some string with spaces](#){: .p-1 .text-blue-100 .fw-700}
 "
 >
 > ▪
-[When](#){: .p-1 .text-red-200} 
-its 
-[property](#){: .p-1 .text-green-200 .fw-700} 
+[When](#){: .p-1 .text-red-200}
+its
+[property](#){: .p-1 .text-green-200 .fw-700}
 consists
 "
-[some string with spaces](#){: .p-1 .text-blue-100 .fw-700} 
+[some string with spaces](#){: .p-1 .text-blue-100 .fw-700}
 "
 >
 > ▪
-[When](#){: .p-1 .text-red-200} 
-its 
-[dictionary](#){: .p-1 .text-green-200 .fw-700} 
+[When](#){: .p-1 .text-red-200}
+its
+[dictionary](#){: .p-1 .text-green-200 .fw-700}
 includes an entry where
 "
-[key](#){: .p-1 .text-blue-100 .fw-700} 
+[key](#){: .p-1 .text-blue-100 .fw-700}
 "
 is
 "
-[value](#){: .p-1 .text-blue-100 .fw-700} 
+[value](#){: .p-1 .text-blue-100 .fw-700}
 "
 >
 >
 | key | Description | Examples |
 |:---:|:----------|:-|
-| [property](#){: .p-1 .text-green-200 .fw-700} | any property that resource have (e.g. `name`, `address`, etc. ). Using `address` will give the terraform object name | `address` `name` `size` |
+| [property](#){: .p-1 .text-green-200 .fw-700} | any property that resources have. Using `type` will give the Terraform resource type; `address` is the name you have given it | `address` `name` `size` |
```

Comment:
Very nice addition! Let's also add `index` here then for resources that have `for_each` functions used.

Reply:
Hmm sounds like I wasn't clear enough. `index` can be used instead of `address` for cases that have `for_each`, in case user would like to create more granular scenarios.

Reply:
Just to check on the usage, you'd write something like:

```
When its name is my_mx_records
And its index is 2
```

Which would address the 3rd item in a list of MX records generated from a loop?

Reply:
On `for_each` terraform requires to have a `key:value` map where it will iterate over the elements and transform a resource address, `my_mx_records` in your example to `my_mx_records["my_for_each_key"]` where users can access to this `key:value` map via `each.key` or `each.value` within the resource.

In this case `index` mapped to `my_for_each_key`.

Reply:
Ah just seen your other comment; I wonder if there's space to explain all that in the table. Could reference `index` there and add a "see below". That may be out of scope for this though as I am less familiar with the styling of other components in the page.

Reply:
That is so true. Let's merge this, I will handle that part.